src/lib.rs

```diff
@@ -53,3 +49,27 @@ impl Behaviour {
         }
     }
 }
+
+pub async fn load_secret_key_from_file(path: impl AsRef<Path>) -> Result<SecretKey> {
+    let bytes = fs::read(path).await?;
+    let secret_key = SecretKey::from_bytes(bytes)?;
+    Ok(secret_key)
+}
+
+#[derive(Debug, thiserror::Error)]
+#[error("Could not generate secret file at specified location. File {path} already exists")]
+pub struct FileAlreadyExistsAtSpecifiedLocation {
```

Comment:
Unless you want to downcast to this error somewhere, I would probably not bother making a dedicated type.

You can achieve the same thing with `.context`!

Reply:
removed this. Not needed with the new tokio::fs API's